src/main/java/io/fabric8/maven/docker/config/handler/property/PropertyConfigHandler.java

```diff
@@ -155,21 +160,23 @@ private String extractName(String prefix, Properties properties) throws IllegalA
         return name;
     }
 
+    private static final Function<String[], String> EXTRACT_SECOND = new Function<String[], String>() {
+        @Override
+        public String apply(String[] p) {
+            return p[1];
+        }
+    };
+
     // Extract only the values of the port mapping
     private List<String> extractPortValues(String prefix, Properties properties) {
-        List<String> ret = new ArrayList<>();
         List<String> ports = listWithPrefix(prefix, PORTS, properties);
         if (ports == null) {
             return null;
         }
-        List<String[]> parsedPorts = EnvUtil.splitOnLastColon(ports);
-        for (String[] port : parsedPorts) {
-            ret.add(port[1]);
-        }
-        return ret;
+        Iterable<String[]> parsedPorts = EnvUtil.splitOnLastColon(ports);
+        return Lists.newArrayList(Iterables.transform(parsedPorts, EXTRACT_SECOND));
```

Comment:
All in all this is not much smaller than the first, more direct version for extracting the port ;-) As much as I appreciate higher abstractions were necessary, in this case I believe the original version is more readable (and does not require knowledge of guava)